packages/store/addon/-private/system/record-array-manager.js

```diff
@@ -11,6 +11,25 @@ import { RecordArray, AdapterPopulatedRecordArray } from './record-arrays';
 import { internalModelFactoryFor } from './store/internal-model-factory';
 
 const emberRun = emberRunloop.backburner;
+const pendingForIdentifier = new Set([]);
+
+function recordArraysForIdentifier(store, identifier) {
+  let internalModel = internalModelFactoryFor(store).peek(identifier);
+  if (internalModel) {
```

Comment:
There was only one failing test that required this guard.  `disconnectRecord` in `store-wrapper-test`.  It is pushing with `data: []` and `included: [houseHash]`.  Is this possible?

Reply:
Turns out M3 hit this case so I had to make some adjustments, roughly speaking it is possible in the current state of things to try to sync-destroy during push via this API. I'm going to bring this up in a meeting as EmberData does not do a sync-destroy here itself and we likely should not allow it going forward as the "syncLiveArray" code is undesirable complexity that remains from our need to handle observers greedily firing when we begin changes.

We may be able to refactor overtime to avoid delivering changes to the UI until a single "ui flush" which would help us avoid this problem.